src/client/Microsoft.Identity.Client/AppConfig/ConfidentialClientApplicationBuilder.cs

```diff
@@ -97,6 +97,55 @@ public ConfidentialClientApplicationBuilder WithCertificate(X509Certificate2 cer
             return this;
         }
 
+
+        /// <summary>
+        /// Sets the certificate associated with the application
+        /// </summary>
+        /// <param name="certificate">The X509 certificate used as credentials to prove the identity of the application to Azure AD.</param>
+        /// <param name="withSendX5C"><c>true</c> if the x5c should be sent. Otherwise <c>false</c>.
+        /// The default is <c>false</c></param>
+        /// <remarks>You should use certificates with a private key size of at least 2048 bytes. Future versions of this library might reject certificates with smaller keys. </remarks>
+        public ConfidentialClientApplicationBuilder WithCertificate(X509Certificate2 certificate, bool withSendX5C)
```

Comment:
`WithClientClaims` also needs to be modified to take in this flag.

Reply:
So, I wasn't sure if WithClientClaims could itself support injecting the x5c claim. I was worried that we might collide. If it only injects claims into the body, which makes sense, then yeah, for sure.